packages/xchain-bitcoincash/src/client.ts

```diff
@@ -0,0 +1,331 @@
+   * @returns {void}
+   *
+   * @throws {"Network must be provided"}
+   * Thrown if network has not been set before.
+   */
+  setNetwork = (network: Network): void => {
+    if (!network) {
+      throw new Error('Network must be provided')
+    } else {
+      this.network = network
+    }
+  }
+
+  /**
+   * Get the current network.
+   *
+   * @returns {Network} The current network. (`mainnet` or `testnet`)
+   */
+  getNetwork = (): Network => {
+    return this.network
+  }
+
+  /**
+   * Get DerivePath
+   *
+   * @returns {string} The bitcoin cash derivation path based on the network.
+   */
+  derivePath(): string {
+    const { testnet, mainnet } = utils.getDerivePath()
+    return utils.isTestnet(this.network) ? testnet : mainnet
+  }
+
+  /**
+   * Get the explorer url.
+   *
+   * @returns {string} The explorer url based on the network.
+   */
+  getExplorerUrl = (): string => {
+    const networkPath = utils.isTestnet(this.network) ? 'tbch' : 'bch'
+    return `https://explorer.bitcoin.com/${networkPath}`
+  }
+
+  /**
+   * Get the explorer url for the given address.
+   *
+   * @param {Address} address
+   * @returns {string} The explorer url for the given address based on the network.
+   */
+  getExplorerAddressUrl = (address: Address): string => {
+    return `${this.getExplorerUrl()}/address/${address}`
+  }
+
+  /**
+   * Get the explorer url for the given transaction id.
+   *
+   * @param {string} txID The transaction id
+   * @returns {string} The explorer url for the given transaction id based on the network.
+   */
+  getExplorerTxUrl = (txID: string): string => {
+    return `${this.getExplorerUrl()}/tx/${txID}`
+  }
+
+  /**
+   * @private
+   * Get private key.
+   *
+   * Private function to get keyPair from the this.phrase
+   *
+   * @param {string} phrase The phrase to be used for generating privkey
+   * @returns {PrivateKey} The privkey generated from the given phrase
+   * */
+  private getPrivateKey = (phrase: string): bitcash.PrivateKey => {
+    const derive_path = this.derivePath()
+    const mnemonic = new Mnemonic(phrase)
+    const hdPrivKey: bitcash.HDPrivateKey = mnemonic.toHDPrivateKey().derive(derive_path)
+
+    return bitcash.PrivateKey.fromObject(hdPrivKey.privateKey)
+  }
+
+  /**
+   * Get the current address.
+   *
+   * Generates a network-specific key-pair by first converting the buffer to a Wallet-Import-Format (WIF)
+   * The address is then decoded into type P2WPKH and returned.
+   *
+   * @returns {Address} The current address.
+   *
+   * @throws {"Phrase must be provided"} Thrown if phrase has not been set before.
+   * @throws {"Address not defined"} Thrown if failed creating account from phrase.
+   */
+  getAddress = (): Address => {
+    if (this.phrase) {
+      const privKey = this.getPrivateKey(this.phrase)
+      const pubKey = bitcash.PublicKey.fromPrivateKey(privKey)
+      const address = bitcash.Address.fromPublicKey(pubKey, utils.bchNetwork(this.getNetwork()))
```

Comment:
Check if any of those function might fail by throwing an error. For example: `bitcash.PublicKey.fromPrivateKey` [might throw an error](https://github.com/bitpay/bitcore/blob/fb04388e4c18d1c8eb2e84496ed28c62af243017/packages/bitcore-lib-cash/test/publickey.js#L212-L216)